### PR TITLE
Support .pyw files, explore a solution for #1160

### PR DESCRIPTION
**Pull request by devdanzin**
This PR changes a few places in logic.py that expected .py files to also accept .pyw files, allowing opening, checking, and tidying (saving already worked).

[The first commit](https://github.com/mu-editor/mu/commit/f9ce9d630aa4c2514764e7ae5b67546ce77a11a4#diff-50021fc2523f19f6dbad358e978343012cd08531c4675a8f087973a5547220ee) just duplicates the checks, but I refactored that into a helper method. 

Regardless of whether this will be the approach we take, we need a design for tests: lots of duplication? Check both extensions in cases only .py was tested? Just a couple specific .pyw tests?

### CODE REVIEW

File: mu/logic.py [hunk 1]
```diff
@@ -1204,11 +1205,12 @@ def load(self, *args, default_path=None):
...
         """
         # Get all supported extensions from the different modes
-        extensions = ["py"]
+        extensions = [ext.strip("*.") for ext in self.python_extensions]
         for mode_name, mode in self.modes.items():
             if mode.file_extensions:
                 extensions += mode.file_extensions
-        extensions = set([e.lower() for e in extensions])
+        # Sort extensions to get deterministic output
+        extensions = list(sorted(set([e.lower() for e in extensions])))
```
Comment (dybber) [inline mu/logic.py:1213]: I guess you don't need set() here?

File: mu/logic.py [hunk 2]
```diff
@@ -1859,3 +1863,13 @@ def tidy_code(self):
                 "these problems."
             )
             self._view.show_message(message, information)
+
+    def has_python_extension(self, filename):
+        """
+        Check whether the given filename matches recognized Python extensions.
+        """
+        file_ends = filename.lower().endswith
+        for ext in self.python_extensions:
```
Comment (dybber) [inline mu/logic.py:1872]: Explicit loop can be avoid using any()

File: mu/logic.py [hunk 3]
```diff
@@ -1303,7 +1305,9 @@ def check_for_shadow_module(self, path):
         return False.
         """
         logger.info('Checking path "{}" for shadow module.'.format(path))
-        filename = os.path.basename(path).replace(".py", "")
+        filename = (
+            os.path.basename(path).replace(".pyw", "").replace(".py", "")
```
Comment (dybber) [inline mu/logic.py:1309]: Maybe we should actually do a case-insensitive replace here? Not sure it's worth the extra work though.

**Review by dybber — APPROVED** [review]
Looks good! I added a few suggestion, but nothing important. Didn't know the modes already had a file_extensions field.

File: mu/logic.py (hunk 1 above)
Comment (devdanzin) [reply mu/logic.py:1213]: As things stand, no, using a set() here doesn't help at all and introduces unstable ordering. It can be argued that it might help when adding modes, but IMO it'd need a test exercising extension deduplication to justify the set() staying.

Having the results sorted or ordered seems desirable, removing the set and sort gives us order. I'll try it like that.

File: mu/logic.py (hunk 2 above)
Comment (devdanzin) [reply mu/logic.py:1872]: Sweet!

File: mu/logic.py (hunk 3 above)
Comment (devdanzin) [reply mu/logic.py:1309]: Makes sense, we accept uppercase extensions so IMO we should do case-insensitive replace now rather than wait for a user to hit this corner case. Trying it.

**Comment by devdanzin** [comment]
Thanks for the review, @dybber! Changes pushed. 

Should we add a couple of new tests for these changes?

**Comment by dybber** [comment]
Great, I don't think we need any tests of these small changes
